lib/superagent-mock.js

```diff
@@ -47,6 +47,45 @@ function mock (superagent, config, logger) {
   }();
 
   /**
+   * Chains progrss events and eventually returns the response
+   * @param ProgressEvent (object/function) the ProgressEvent implementation
+   * @param request (object) the current request
+   *
+   * @param progress (object) the progress configuration
+   * @param progress.parts (number)
+   * @param progress.delay (number)
+   * @param progress.total (number)
+   * @param progress.lengthComputable (boolean)
+   * @param progress.direction (string)
+   *
+   * @param remainingParts (number) the remaining calls to make
+   * @param callback (function) the end response callback
+   */
+  var handleProgress = function (ProgressEvent, request, progress, remainingParts, callback) {
+    setTimeout(function () {
+      var total = progress.total || 100;
+      var loaded = total * (progress.parts - remainingParts) / progress.parts;
+      var lengthComputable = typeof progress.lengthComputable !== 'undefined' ? !!progress.lengthComputable : true;
+      var e = ProgressEvent.initProgressEvent
+        ? ProgressEvent.initProgressEvent('', true, false, lengthComputable, loaded, total) // IE10+
+        : new ProgressEvent('', {lengthComputable: lengthComputable, loaded: loaded, total: total});
+
+      if (e.total > 0) {
+        e.percent = e.loaded / e.total * 100;
+      }
+      e.direction = progress.direction || 'upload';
+
+      request.emit('progress', e);
+
+      if (remainingParts > 0) {
+        handleProgress(ProgressEvent, request, progress, remainingParts - 1, callback);
+      } else {
+        callback();
+      }
+    }, progress.delay || 0);
```

Comment:
You should check [this](http://www.2ality.com/2015/06/tail-call-optimization.html), `setTimeout` is definitely not required